model/riscv_insts_zicboz.sail

```diff
@@ -0,0 +1,64 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+// Cache Block Operations - Zero
+
+function cbo_zero_enabled(p : Privilege) -> bool = feature_enabled_for_priv(p, menvcfg[CBZE][0], senvcfg[CBZE][0])
+
+/* ****************************************************************** */
+union clause ast = RISCV_ZICBOZ : (regidx)
+
+mapping clause encdec = RISCV_ZICBOZ(rs1)                if haveZicboz()
+  <-> 0b000000000100 @ rs1 @ 0b010 @ 0b00000 @ 0b0001111 if haveZicboz()
+
+mapping clause assembly = RISCV_ZICBOZ(rs1)
+  <-> "cbo.zero" ^ spc() ^ "(" ^ opt_spc() ^ reg_name(rs1) ^ opt_spc() ^ ")"
+
+function clause execute(RISCV_ZICBOZ(rs1)) = {
+  if cbo_zero_enabled(cur_privilege) then {
+    let rs1_val = X(rs1);
+    let cache_block_size_exp = plat_cache_block_size_exp();
+    let cache_block_size = 2 ^ cache_block_size_exp;
+
+    // Offset from rs1 to the beginning of the cache block. This is 0 if rs1
+    // is aligned to the cache block, or negative if rs1 is misaligned.
+    let offset = (rs1_val & ~(zero_extend(ones(cache_block_size_exp)))) - rs1_val;
+
+    // TODO: This is not correct since CHERI treats CBO operations
+    // differently to normal accesses with respect to bounds checks.
+    match ext_data_get_addr(rs1, offset, Write(Data), cache_block_size) {
```

Comment:
```suggestion
    match ext_data_get_addr(rs1, offset, Write(Data), cache_block_size) {
```
CBOZ should be correct for CHERI since we check all bytes?